core/client/src/light/fetcher.rs

```diff
@@ -108,6 +108,15 @@ pub struct ChangesProof<Header: HeaderT> {
 	pub roots_proof: Vec<Vec<u8>>,
 }
 
+/// Remote block body request
+#[derive(Clone, Default, Debug, PartialEq, Eq, Hash)]
+pub struct RemoteBodyRequest<Header: HeaderT> {
+	/// Header of the requested block body
+	pub header: Header,
```

Comment:
We probably only need the hash.

Reply:
The header could be pruned (replaced with CHT) when request+response messages are routing from/to light node => to check it we'll need to retrieve header again from the full node. Holding header in the request guarantees that we can verify response without additional messages (we need to check extrinsics root for that).

Reply:
Oh, I see that @niklasad1 is also fetching header here. So probably you're right